File: src/main/java/ordilov/randomplay/playlist/domain/youtube/YoutubePlaylistItem.java
package ordilov.randomplay.playlist.domain.youtube;

import lombok.Getter;
import ordilov.randomplay.playlist.domain.youtube.Snippet.PlaylistItem;

@Getter
public class YoutubePlaylistItem {
  private String id;
  private String kind;
  private String etag;
  private Status status;
  private PlaylistItem snippet;
  private ContentDetails contentDetails;

  @Getter
  private static class ContentDetails{
    private String note;
    private String endAt;
    private String videoId;
    private String startAt;
  }

  @Getter
  private static class Status{
    private String privacyStatus;
  }
}

File: src/main/java/ordilov/lissn/playlist/domain/youtube/YoutubePlaylistItem.kt
package ordilov.lissn.playlist.domain.youtube

data class YoutubePlaylistItem(
    val id: String,
    val kind: String,
    val etag: String,
    val status: Status,
    val snippet: PlaylistItem,
    val contentDetails: ContentDetails,
) {
    data class ContentDetails(
        val note: String,
        val endAt: String,
        val videoId: String,
        val startAt: String,
    )

    data class Status(
        val privacyStatus: String
    )
}